### DISCUSSION TITLE
Update improves on my Alienware m15 Ryzen Edition R5

### DISCUSSION THREAD

**Discussion by Burnrate** [T-Troll/alienfx-tools#245] [General]
I have an Alienware m15 Ryzen Edition R5 laptop I bought November of last year (2021).  AWCC and dell support has truly been a trainwreck of course.  There is nothing I can do in the bios or system setup to alter the fans.  This program is a lifesaver.

With the latest update I can use level 2 to get my fans [1] to 5100 RPM and [2] to 5300 RPM.  I know they have gone up to 6000 RPM before but I can't seem to get it there.  Using the get max functionality will only let the manual mode push the fans to 4700 RPMs which is better than the previous version that put it up to 4000 RPM.  Using G mode will bring the fans down below 4k RPM and Level one has them around 4300 RPM.

I can use the manual mode in non demanding situations but lots of the time I just leave it on level 2 so I don't get tons of cpu throttling from the heat.

Either way I just wanted to say thanks for this awesome option and let you know what I was experiencing just in case it helps anything.

**Comment by T-Troll** [comment]
Being honest, i don't change anything into power modes for this release.
Only broke manual g-mode switch (see [7.6.0.1 Hotfix](https://github.com/T-Troll/alienfx-tools/releases/tag/7.6.0.1)).

But seems like your system have dedicated PM for G-mode, like some G5's.
Can you, please, help me to diagnose it? For doing so:
- Edit all power mode names (it's needed to insert it's IDs into configuration) - Manual and G-mode can't be edited, it's OK.
- Save configuration or quit app.
- Backup configuration (using `alienfx-config backup` command from CMD).
- Share `afx-fans.reg` here (it doesn't have any personal or system data).
- Open Administrator CMD at app folder, run `alienfan-cli rpm`. At the second line of the output, you can find Hardware ID (it's into "Supported hardware (XXXX)..."). Share this XXXX with me.

In case i right, i'll add your system to exception set mode, and G-mode will work better.

**Comment by Burnrate** [comment]
For some reason I cannot edit or create any power plans.  Maybe a windows 11 bug.

Got a couple errors from the backup
![image](https://user-images.githubusercontent.com/2849955/202589184-9799b216-3e11-4164-bd46-b2a660b28707.png)

This is the afx-fans.reg file:
`Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Alienfan]
"StartAtBoot"=dword:00000001
"StartMinimized"=dword:00000000
"LastPowerStage"=dword:00000002
"UpdateCheck"=dword:00000001
"ObCheck"=dword:00000001
"DisableAWCC"=dword:00000001
"KeyboardShortcut"=dword:00000001
"Boost-0"=hex:6c,88,13
"Boost-1"=hex:64,a8,16

[HKEY_CURRENT_USER\Software\Alienfan\Powers]
"Power-0"="Manual"
"Power-151"="First Level"
"Power-153"="Second Level"

[HKEY_CURRENT_USER\Software\Alienfan\Sensors]
"Fan-0-257"=hex:00,00,46,32,59,64,64,64

`
rpm output shows Supported hardware 3200
![image](https://user-images.githubusercontent.com/2849955/202589314-8104a22b-6fdd-4f33-90e6-0871a60bbead.png)

**Reply by T-Troll** [reply]
> For some reason I cannot edit or create any power plans. Maybe a windows 11 bug.

Do you use 7.6.0.1? Let me check anyway.

> Got a couple errors from the backup

It's OK - if you never use other tools - you don't have its configuration. I see you don't run any Light control or monitoring yet.

> This is the afx-fans.reg file:

You can attach it, but this is ok too.

I see you have common power modes, not one dedicated to G-mode, but let me check if i have your BIOS dump - i'll check how system do it.
And yes, your Hardware ID is 3200 - it's the same for all Ryzen R5, i use it to apply some system-specific tricks.

UPD: I check your BIOS - yes, you have same trick as some G5 - G-mode also use 2nd power mode! I'll add a patch at the next release.

**Reply by T-Troll** [reply]
Here we are. I update 7.6.0.1, fixing Power mode name edit and adding G-mode trick (as in BIOS) for your system.
